src/bscPlugin/transpile/BrsFilePreTranspileProcessor.ts

```diff
@@ -178,6 +181,186 @@ export class BrsFilePreTranspileProcessor {
         }
     }
 
+    private processNullCoalescingExpression(nullCoalescingExpression: NullCoalescingExpression, visitor: ReturnType<typeof createVisitor>, walkMode: WalkMode) {
+        // Check if this null coalescing expression has mutating expressions that require scope protection
+        const consequentInfo = util.getExpressionInfo(nullCoalescingExpression.consequent, this.event.file);
+        const alternateInfo = util.getExpressionInfo(nullCoalescingExpression.alternate, this.event.file);
+        
+        let hasMutatingExpression = [
+            ...consequentInfo.expressions,
+            ...alternateInfo.expressions
+        ].find(e => isCallExpression(e) || isCallfuncExpression(e) || isDottedGetExpression(e));
```

Comment:
I don't think mutating expressions actually matter for this feature. If it's an assignment statement, we'll only ever evaluate them exactly once. Consider:
```
value = one() ?? two()
```

That should transpile to
```
value = one()
if value = invalid
    value = two()
end if
```

Notice how there was no duplication? So mutations don't matter as long as we only run them once, and they are only evaluated when the previous expression evaluates to invalid.

For now, like mentioned in other comments, let's only do this logic if it's an assignmentStatement to a local variable (i.e. not a DottedSet or IndexedSet).